docs/java/features/odata/generate-client.md

```diff
@@ -0,0 +1,270 @@
+---
+id: generate-java-odata-vmd-v2-v4
+title: OData VDM Generator for Java
```

Comment:
My understanding of `VDM` is that it is about the SAP S/4HANA only. All other OData clients are just OData clients. Also, if you never heard `VDM` before, this article might not catch your eye if you are looking for the generator.

Reply:
Yeah, I took the term `VDM` to refer to the generated classes in general to have a simple way of referring to them at all places. But you are right we should use the term properly. 

I am a bit hesitant to just write `OData Client Generator`, maybe `Generator for OData Client Libraries`?

Reply:
VDM is our abstraction so if we generate a client reference service it's still VDM. That's one of the topics we wanted to document in generic concepts and refer to. Should I quickly make an initial article there?

Reply:
Quoting from one of our [blog posts](https://blogs.sap.com/2017/09/11/step-10-with-sap-s4hana-cloud-sdk-virtual-data-model-for-odata/):
> The data stored in an S/4HANA system is inherently complexly structured and therefore difficult to query manually. Therefore, HANA introduced a Virtual Data Model (VDM) that aims to abstract from this complexity and provide data in a semantically meaningful and easy to consume way.

So I think Marika is right.